File: Data Structures & Algorithms/jump-game/submission-1.cs
public class Solution {
    public bool CanJump(int[] nums) {
        int maxReach = 0;

        for (int i = 0; i < nums.Length; i++) {
            if (i > maxReach) return false; // can't reach this index
            maxReach = Math.Max(maxReach, i + nums[i]); // update reach
        }

        return true;
    }
}

File: Data Structures & Algorithms/maximum-subarray/submission-1.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maximum = float('-inf')
        cSum = 0
        for i in nums:
            
            if cSum < 0:
                cSum = 0
            
            cSum += i
            maximum = max(maximum,cSum)
            
        return maximum